tests/ci.js

```diff
@@ -1,181 +1,204 @@
-  shell.echo('### stderr');
-  shell.echo(result.stderr || '(no stderr)');
-  shell.echo();
-  shell.echo('### message');
-  shell.echo(message);
-  shell.echo(`Exit code: ${result.code}. See stdout and stderr above.`);
-  shell.echo(`This message comes from tests/${filename}`);
-  shell.echo();
-  shell.exit(1);
+function assertTestErrored(runResult) {
+  const msg = 'Expected error (exit code ' + resultErrored + '), but got ' + runResult.status;
+  assert.strictEqual(resultErrored, runResult.status, getDetailedMessage(msg, runResult));  
 }
 
-shell.echo(filename + ': Uninstalling old elm-test...');
-shell.exec('npm remove --ignore-scripts=false --global ' + elmTest);
-
-shell.echo(filename + ': Installing elm-test...');
-shell.exec('npm link --ignore-scripts=false');
-
-var interfacePath = require('elmi-to-json').paths['elmi-to-json'];
-
-shell.echo(filename + ': Verifying elmi-to-json is installed...');
-var interfaceResult = spawn.sync(interfacePath, ['--help']);
-var interfaceExitCode = interfaceResult.status;
-
-if (interfaceExitCode !== 0) {
-  shell.echo(
-    filename +
-      ': Failed because `elmi-to-json` is present, but `elmi-to-json --help` returned with exit code ' +
-      interfaceExitCode
-  );
-  shell.echo(interfaceResult.stdout.toString());
-  shell.echo(interfaceResult.stderr.toString());
-  shell.exit(1);
+function assertTestFailure(runResult) {
+  const msg = 'Expected failure (exit code >= ' + resultFailureThreshold + '), but got ' + runResult.status;
+  assert.ok(runResult.status >= resultFailureThreshold, getDetailedMessage(msg, runResult));  
 }
 
-shell.exec('npm link --ignore-scripts=false');
-
-shell.echo(filename + ': Verifying installed elm-test version...');
-var versionRun = shell.exec(elmTest + ' --version');
-
-if (versionRun.code !== 0) {
-  shell.exec(
-    'echo Expected elm-test --version to exit with exit code 0, but it was ' +
-      versionRun.code
-  );
-  shell.exit(1);
-}
+describe('--help', () => {
+  it('Should exit indicating success', () => {
+    const runResult = execElmiToJSON(['--help']);
+    assertTestSuccess(runResult);
+  }).timeout(60000);
+
+  it('Should print the usage', () => {
+    const runResult = execElmiToJSON(['--help']);
+    // ensure we have a non-empty output
+    assert.ok(runResult.stdout.length > 0);
+  }).timeout(60000);
+});
 
-if (versionRun.stdout.trim() !== elmTestVersion) {
-  shell.exec(
-    'echo Expected elm-test --version to output ' +
-      elmTestVersion +
-      ', but it was ' +
-      versionRun.stdout.trim()
-  );
-  shell.exit(1);
-}
+// shell.exec('npm link --ignore-scripts=false');
+describe('--version', () => {
+  it('Should exit indicating success', () => {
+    const runResult = execElmTest(['--version']);
+    assertTestSuccess(runResult);
+  }).timeout(60000);
+
+  it('Should print the usage', () => {
+    const runResult = execElmTest(['--version']);
+    assert.strictEqual(elmTestVersion, runResult.stdout.trim());
+  }).timeout(60000);
```

Comment:
- Could these be merged into one? `Should print the version and exit with 0`.
- Does this really need a 1 minute timeout? If `elm-test --version` takes one minute we must have done something very wrong?

Reply:
That's true - I took the timeout(60000); from flags.js, where it runs the tests for a whole package. I've set it to 5 seconds and merged both tests into one.

Reply:
> If `dummyBinPath` is unused, can’t you change the code to:
> 
> ```
> const { fixturesDir, spawnOpts } = require('./util');
> ```
> 
> ?

I had assumed that the number of assignment targets has to match the number of exports from util.js. Fixed, just another indicator of my very limited Javascript knowledge :-)